src/optimizer/build_probe_side_optimizer.cpp

```diff
@@ -0,0 +1,135 @@
+#include "duckdb/optimizer/build_probe_side_optimizer.hpp"
+#include "duckdb/planner/operator/logical_join.hpp"
+#include "duckdb/execution/physical_plan_generator.hpp"
+#include "duckdb/planner/operator/logical_any_join.hpp"
+#include "duckdb/planner/operator/logical_comparison_join.hpp"
+#include "duckdb/common/enums/join_type.hpp"
+
+namespace duckdb {
+
+BuildProbeSideOptimizer::BuildProbeSideOptimizer(ClientContext &context, LogicalOperator &op) : context(context) {
+	vector<ColumnBinding> updating_columns, current_op_bindings;
+	if (op.type == LogicalOperatorType::LOGICAL_UPDATE) {
```

Comment:
Rather than trying to reconstruct that this is an update (or delete, which suffers from the same problem but isn't handled here) - it might be easier to look at the `LogicalGet` and check if there is a `rowid` being projected in the set of `column_ids`. This is generally only the case for updates/deletes (or when the user manually projects `rowid` - but this is much less common).